docs/designs/allreduce_based_horovod.md

```diff
@@ -0,0 +1,117 @@
+# Design for Elastic AllReduce-based Training Support Based on Horovod
+
+This document describes the design for supporting AllReduce-based distributed
+training based on Horovod in ElasticDL.
+
+## Motivation
+
+We have developed elastic AllReduce based on FTlib in ElasticDL.
+From the [benchmark report](../benchmark/ftlib_benchmark.md), we can
+find that the performance of FTlib for ResNet50 is worse than Horovod.
+What's more, it is not stable to build consensus by gossip protocol in FTlib
+like the [issue](https://github.com/sql-machine-learning/elasticdl/issues/2192#issuecomment-664096185).
+
+FTlib uses Gloo to implement elastic AllReduce because the worker process
+can catch the exception from Gloo and not exit if the AllReduce operator
```

Comment:
Does AllReduce operator here means `Broadcast`/`AllReduce`/`AllGather`? It's different from Kubernetes operator.

Reply:
FTlib uses the gossip protocol to build consensus, which is not stable as described in this issue(link).

Reply:
if any collective communications fail.